### ISSUE TITLE
Operations with differently sized DenseVectors where `a + b` fails but `b + a` succeeds

### ISSUE THREAD

**Issue by adgaudio** [scalanlp/breeze#515]
Hello!

I came across this oddity and was wondering if it's expected behavior or a bug?  Specifically, we can perform operations with DenseVectors of different sizes if the small vector comes first.

```
scala> val a = DenseVector(1, 2, 3, 4, 5)
a: breeze.linalg.DenseVector[Int] = DenseVector(1, 2, 3, 4, 5)

scala> val b = DenseVector(1, 2, 3)
b: breeze.linalg.DenseVector[Int] = DenseVector(1, 2, 3)

scala> b + a
res71: breeze.linalg.DenseVector[Int] = DenseVector(2, 4, 6)

scala> a + b
java.lang.ArrayIndexOutOfBoundsException: 3
  at breeze.linalg.operators.DenseVectorOps$$anon$142.apply(DenseVectorOps.scala:161)
  at breeze.linalg.operators.DenseVectorOps$$anon$142.apply(DenseVectorOps.scala:150)
  at breeze.linalg.NumericOps$class.$plus(NumericOps.scala:172)
  at breeze.linalg.DenseVector.$plus(DenseVector.scala:51)
  ... 42 elided

```

It does seem like there should be validation that causes `b + a` to fail, though if we fixed it I bet we'd break people's downstream code that depends on this condition.  

If you want a PR for a "fix" despite my previous comment about backwards incompatibility, I'll be happy to work on this in a week or two (and please provide a couple tips where I should include the validation!).

Thanks for reviewing this!

**Comment by adgaudio** [comment]
update: this doesn't work for double:

```
scala> DenseVector(1D, 2D, 3D) + DenseVector(1D, 2D, 3D, 4D, 5D)
java.lang.IllegalArgumentException: requirement failed: Vectors must have same length: x.length == y.length (5 != 3)
  at breeze.linalg.DenseVector$canDaxpy$.apply(DenseVector.scala:589)
  at breeze.linalg.DenseVector$$anon$3.apply(DenseVector.scala:581)
  at breeze.linalg.DenseVector$$anon$3.apply(DenseVector.scala:579)
  at breeze.linalg.operators.DenseVector_GenericOps$$anon$308.apply(DenseVectorOps.scala:710)
  at breeze.linalg.operators.DenseVector_GenericOps$$anon$308.apply(DenseVectorOps.scala:707)
  at breeze.linalg.NumericOps$class.$plus(NumericOps.scala:172)
  at breeze.linalg.DenseVector.$plus(DenseVector.scala:51)
  ... 42 elided
```

**Comment by dlwh** [comment]
that should be a failure. please submit a PR!

**Comment by dlwh** [comment]
(I don't care about backwards compatibility for this. we're pre 1.0 for now, and it's a clear bug)

**Comment by adgaudio** [comment]
Sounds great - I plan to start working on a PR either next week or in 2 weeks, and will link to this issue when I post the PR.

**Comment by dlwh** [comment]
awesome!

**Comment by adgaudio** [comment]
fixed in #528
